ghc-toolkit/boot-libs/integer-simple/GHC/Integer/Type.hs

```diff
@@ -234,9 +234,9 @@ foreign import javascript "__asterius_jsffi.Integer.smallInteger(${1}, ${2})" js
 
 foreign import javascript "__asterius_jsffi.Integer.wordToInteger(${1}, ${2})" js_wordToInteger :: Word# -> Word# -> Int#
 
-foreign import javascript "__asterius_jsffi.Integer.integerToWord(${1})" js_integerToWord :: Int# -> Word#
+-- | Given integer and which 32-bit _piece_ of the word we want, return that piece. 
+foreign import javascript "__asterius_jsffi.Integer.integerToWord(${1}, ${2})" js_integerToWord :: Int# -> Int# -> Word#
 
-foreign import javascript "__asterius_jsffi.Integer.integerToInt(${1})" js_integerToInt :: Int# -> Int#
```

Comment:
Can we do the same for `integerToWord`/`integerToInt` and not rely on `int2Word#` or `word2Int#`? Won't be too much work and I sense some danger in the coercing.